src/printer.js

```diff
@@ -3831,6 +3845,7 @@ function nodeStr(node, options) {
   const alternate = preferred === single ? double : single;
 
   let shouldUseAlternateQuote = false;
+  let canChangeDirectiveQuotes = !rawContent.includes('\\');
```

Comment:
You're right, we should ideally change to the preferred quote if possible. However, since changing the quote involves calling `makeString`, it could remove some of the unnecessary escapes, which would change the meaning of the directive (see https://github.com/prettier/prettier/pull/1560#discussion_r115609519). So, I just used the presence of any backslash at all as an indicator to leave the entire thing alone.

Hopefully, this is enough of an edge case that no one will actually run into it, as you mentioned in https://github.com/prettier/prettier/pull/1560#issuecomment-300296221. I'd still like to get it merged with the tests passing, though, so that we can set `AST_EXPLORE=1` on Travis to more rigorously test correctness.